From 7b2603f9dab962873df970eeadc4e323d113bda5 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Thu, 29 Sep 2016 10:34:12 +0300
Subject: [PATCH 0601/1345] dts: opp: ap806: add operating-points-v2
 description for AP806 cpus

commit  821ec2e27901c142f840e0b90031a58c09ef52ab from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Operating Performance Point (OPP) are voltage-current-frequency
combinations.
Added operating-points-v2 description for AP806 cpus:
- 2 Clusters, each with 2 shared cpus
- single OPP (Operating Performance Points) of 100Mhz per cluster.

AP806 DFS support only full integer dividers, so only a single
OPP (100Mhz) is added statically in device tree, which is suited
for all CPU Sample at reset modes.
Additional OPP entries will be registered at runtime by cpufreq driver,
according to actual booted CPU frequency.

Change-Id: If3b5078b93ece357fe3c8f7bd1426ff44c6e8909
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33199
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi |    2 ++
 arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi |    4 ++++
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi      |   21 ++++++++++++++++++++
 3 files changed, 27 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi
index 8b405e9..25b95cf 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806-dual.dtsi
@@ -59,6 +59,7 @@
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x000>;
 			clocks = <&ap806_clock 0>;
+			operating-points-v2 = <&cluster0_opp>;
 			enable-method = "psci";
 		};
 		cpu@001 {
@@ -66,6 +67,7 @@
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x001>;
 			clocks = <&ap806_clock 0>;
+			operating-points-v2 = <&cluster0_opp>;
 			enable-method = "psci";
 		};
 	};
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi
index 6fd612c..75f4fdb 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806-quad.dtsi
@@ -59,6 +59,7 @@
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x000>;
 			clocks = <&ap806_clock 0>;
+			operating-points-v2 = <&cluster0_opp>;
 			enable-method = "psci";
 		};
 		cpu1: cpu@001 {
@@ -66,6 +67,7 @@
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x001>;
 			clocks = <&ap806_clock 0>;
+			operating-points-v2 = <&cluster0_opp>;
 			enable-method = "psci";
 		};
 		cpu2: cpu@100 {
@@ -73,6 +75,7 @@
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x100>;
 			clocks = <&ap806_clock 1>;
+			operating-points-v2 = <&cluster1_opp>;
 			enable-method = "psci";
 		};
 		cpu3: cpu@101 {
@@ -80,6 +83,7 @@
 			compatible = "arm,cortex-a72", "arm,armv8";
 			reg = <0x101>;
 			clocks = <&ap806_clock 1>;
+			operating-points-v2 = <&cluster1_opp>;
 			enable-method = "psci";
 		};
 	};
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index c67ed73..adef92d 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -66,6 +66,27 @@
 		method = "smc";
 	};
 
+	/* CPUfreq Cluster settings:
+	 * - Initial required static OPP entry of 100Mhz - suited for all CPU boot options
+	 * - Additional OPP entries are registered at runtime by cpufreq driver
+	 */
+	cluster0_opp: opp_table0 {
+		compatible = "operating-points-v2";
+		opp-shared;
+		opp@0 {
+			opp-hz = /bits/ 64 <100000000>;
+			clock-latency-ns = <50000>;
+		};
+	};
+	cluster1_opp: opp_table1 {
+		compatible = "operating-points-v2";
+		opp-shared;
+		opp@0 {
+			opp-hz = /bits/ 64 <100000000>;
+			clock-latency-ns = <50000>;
+		};
+	};
+
 	ap806 {
 		#address-cells = <2>;
 		#size-cells = <2>;
-- 
1.7.9.5

